packages/fhir-location-management/src/components/EditLink/index.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { useMls } from '../../mls';
+import {
+  BACK_SEARCH_PARAM,
+  URL_LOCATION_UNIT_EDIT,
+  URL_SERVICE_POINT_ADD_EDIT,
+} from '../../constants';
+import { ILocation } from '@smile-cdr/fhirts/dist/FHIR-R4/interfaces/ILocation';
+
+interface EditLinkProps {
+  location: ILocation;
+  editLinkText?: string;
+}
+
+const EditLink: React.FC<EditLinkProps> = (props) => {
+  const { location, editLinkText } = props;
+  const { t } = useMls();
+  const { id, physicalType } = location;
+  const isBuilding = physicalType?.coding?.[0].code === 'bu';
+  const currentLocation = useLocation();
+
+  const backToParam = new URLSearchParams({ [BACK_SEARCH_PARAM]: currentLocation.pathname });
+
+  return (
+    <Link
+      to={
+        isBuilding
+          ? `${URL_SERVICE_POINT_ADD_EDIT}/${id}`
+          : `${URL_LOCATION_UNIT_EDIT}/${id}?${backToParam}`
+      }
+      className="m-0 p-1"
+    >
+      {t(editLinkText ? editLinkText : 'Edit')}
```

Comment:
the translator function `t` value's should be a literal string so that its extractable. I think we can just render the prop and rely on it being already translated in the calling component